File: app/src/main/java/com/github/premnirmal/ticker/widget/RemoteStockProviderService.java
package com.github.premnirmal.ticker.widget;

import android.content.Intent;
import android.widget.RemoteViewsService;

/**
 * Created by premnirmal on 12/21/14.
 */
public class RemoteStockProviderService extends RemoteViewsService {

    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        return new RemoteStockViewAdapter(getApplicationContext());
    }
}

File: app/src/main/kotlin/com/github/premnirmal/ticker/widget/RemoteStockProviderService.kt
package com.github.premnirmal.ticker.widget

import android.appwidget.AppWidgetManager
import android.content.Intent
import android.widget.RemoteViewsService

/**
 * Created by premnirmal on 2/27/16.
 */
class RemoteStockProviderService : RemoteViewsService() {

  override fun onGetViewFactory(intent: Intent): RemoteViewsService.RemoteViewsFactory {
    val appWidgetId =
      intent.getIntExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, AppWidgetManager.INVALID_APPWIDGET_ID)
    return RemoteStockViewAdapter(appWidgetId)
  }
}